src/actions/index.js

```diff
@@ -1,10 +1,22 @@
-import images from '../apis/images';
+import { GetRecent, SearchPhoto } from '../apis/flickr';
 import {
-    FETCH_PHOTO
+    GET_RECENT,
+    SEARCH_PHOTO,
+    ON_TYPE
 } from './types';
 
-export const fetchPhoto = () => async dispatch => {
-    const response = await images.get();
+export const getRecent = () => async dispatch => {
+    const response = await GetRecent.get();
 
-    dispatch({ type: FETCH_PHOTO, payload: response.data.photos.photo});
+    dispatch({ type: GET_RECENT, payload: response.data.photos.photo});
+};
+
+export const searchPhoto = (term) => async dispatch => {
+    const response = await SearchPhoto.get(term);
+
+    dispatch({ type: SEARCH_PHOTO, payload: response.data.photos.photo});
+};
+
+export const onType = (term) => dispatch => {
```

Comment:
Lets think about naming convention here. If I'm looking at this from a 100 foot view, say we have 30 components. What does onType mean? We could be referring to a type change (as in primitive), this could be a reference to an API property, its kinda vague. Can we think of a naming convention that is a bit more explicit? Maybe something like `ON_SEARCH_TERM_KEYSTROKE`, or `HANDLE_SEARCH_TERM_CHANGE`.

Reply:
I'll go with your second suggestion